core/src/modules/ops/profileProperty.ts

```diff
@@ -73,6 +79,64 @@ export namespace ProfilePropertyOps {
         throw new Error(`cannot coerce profileProperty type ${type}`);
     }
   }
+
+  export async function processPendingProfileProperties(
+    property: Property,
+    limit = 100,
+    delayMs = defaultProfilePropertyProcessingDelay
+  ) {
+    if (!delayMs || delayMs < defaultProfilePropertyProcessingDelay) {
+      delayMs = defaultProfilePropertyProcessingDelay;
+    }
+
+    const source = await property.$get("source", { scope: null });
+    if (source.state !== "ready") return;
```

Comment:
maybe `return []` so this always returns and array.

Reply:
https://github.com/grouparoo/grouparoo/pull/1566/commits/525cb2c2cf9a67abf8e699f9b380611f34cf5ab2